src/device/rdk/voice/voice_functions.rs

```diff
@@ -161,27 +162,24 @@ pub fn sendVoiceCommand(audio_file_in: String) -> Result<(), DabError> {
         loop {
             let response = ws_receive(&mut ws_stream).await?;
 
-            // Alexa specific response; listen to "onServerMessage" for "RequestProcessingCompleted".
-            // {"jsonrpc": "2.0", "method": "client.events.onServerMessage", "params": \
-            //   {"xr_speech_avs":{"directive":{"header":{"namespace":"InteractionModel","name":"RequestProcessingCompleted",...},"payload":{}}}}
-            // }
-            // TODO: Handle other voice implementation using "onSessionEnd".
             if response.get("params")
-                .and_then(|params| params.get("xr_speech_avs"))
-                .and_then(|xr_speech_avs| xr_speech_avs.get("directive"))
-                .and_then(|directive| directive.get("header"))
-                .and_then(|header| header.get("name"))
-                .and_then(|name| name.as_str())
-                .map_or(false, |name_str| name_str == "RequestProcessingCompleted") {
+                .and_then(|params| params.get("result"))
+                .and_then(|result| result.as_str())
+                .map_or(false, |name_str| name_str == "success") {
```

Comment:
Maybe check that `"method"` is `"client.events.onSessionEnd"` (as per [example](https://rdkcentral.github.io/rdkservices/#/api/VoiceControlPlugin?id=example-12)) first to be sure the message you received from websocket is actually the event you're waiting for?

Reply:
Added steps to parse json to match "onSessionEnd" and the "result" status.